WrightTools/_group.py

```diff
@@ -178,21 +179,113 @@ def parent(self):
+                pass
+            finally:
+                if hasattr(self, '_tmpfile'):
+                    os.close(self._tmpfile[0])
+                    os.remove(self._tmpfile[1])
+
+    def copy(self, parent=None, name=None, verbose=True):
+        """Create a copy under parent.
+
+        All children are copied as well.
+
+        Parameters
+        ----------
+        parent : WrightTools Collection (optional)
+            Parent to copy within. If None, copy is created in root of new
+            tempfile. Default is None.
+        name : string (optional)
+            Name of new copy at destination. If None, the current natural
+            name is used. Default is None.
+        verbose : boolean (optional)
+            Toggle talkback. Default is True.
+
+        Returns
+        -------
+        Group
+            Created copy.
+        """
+        if name is None:
+            name = self.natural_name
+        if parent is None:
+            new = Group()
+            # attrs
+            for k, v in self.attrs.items():
+                new.attrs[k] = v
+            new.natural_name = name
+            # children
+            for k in self.keys():
+                self[k].copy(new)
+            new.flush()
+            p = new.filepath
+            from ._open import open as wt_open
+            new = wt_open(p)
+        else:
+            # copy
+            self.file.copy(self.name, parent, name=name)
+            if 'item_names' in parent.attrs.keys():
+                new = parent.item_names + (name,)
+                parent.attrs['item_names'] = np.array(new, dtype='S')
+            parent._update_natural_namespace()
+            new = parent[name]
+        # finish
+        if verbose:
+            print('{0} copied to {1}'.format(self.fullpath, new.fullpath))
+        return new
 
     def flush(self):
         """Ensure contents are written to file."""
         self.file.flush()
+
+    def save(self, filepath=None, verbose=True):
+        """Save as root of a new file.
+
+        Parameters
+        ----------
+        filepath : string (optional)
+            Filepath to write. If None, file is created using natural_name.
+        verbose : boolean (optional)
+            Toggle talkback. Default is True
+
+        Returns
+        -------
+        str
+            Written filepath.
+        """
+        # parse filepath
+        if filepath is None:
+            filepath = os.path.join(os.getcwd(), self.natural_name + '.wt5')
+        elif len(os.path.basename(filepath).split('.')) == 1:
+            filepath += '.wt5'
+        filepath = os.path.expanduser(filepath)
+        # copy to new file
+        h5py.File(filepath)
+        new = Group(filepath=filepath)
+        # attrs
+        for k, v in self.attrs.items():
+            new.attrs[k] = v
+        # children
+        for k in self.keys():
+            self[k].copy(new, verbose=True)
+        # finish
+        import time
+        time.sleep(10)
```

Comment:
Why are we sleeping for ten seconds?????

If waiting is needed, please find a way to check how much wait is needed, rather than having a hard coded number.

Hard coded wait times always lose performance, very noticeably here, and either wait way too long on powerful machines, or not long enough on lower spec machines

Also, import in the middle of the function